### PR TITLE
REST API: Avoid unnecessarily preparing item links

### PR DESCRIPTION
**Pull request by dlh01**
Trac ticket: https://core.trac.wordpress.org/ticket/52992

### CODE REVIEW

**Review by anton-vlasenko — CHANGES_REQUESTED** [review]
Could you please resolve the merge conflicts?

File: src/wp-includes/rest-api/endpoints/class-wp-rest-post-types-controller.php [hunk 1]
```diff
@@ -244,16 +244,18 @@ public function prepare_item_for_response( $item, $request ) {
...
+			$response->add_links(
+				array(
+					'collection'              => array(
+						'href' => rest_url( sprintf( '%s/%s', $this->namespace, $this->rest_base ) ),
+					),
+					'https://api.w.org/items' => array(
+						'href' => rest_url( rest_get_route_for_post_type_items( $post_type->name ) ),
+					),
+				)
+			);
```
Comment (spacedmonkey) [inline src/wp-includes/rest-api/endpoints/class-wp-rest-post-types-controller.php]: I think we should take the opportunity to break this logic out to `prepare_links` method.

**Comment by spacedmonkey** [comment]
Missed links. 

https://github.com/WordPress/wordpress-develop/blob/df8c3d0e089f77eb5c6fa971ba4ae9956255893e/src/wp-includes/rest-api/endpoints/class-wp-rest-sidebars-controller.php#L371

https://github.com/WordPress/wordpress-develop/blob/bd08b221273b8b367a18fef8c9efb873819ac89d/src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php#L665

https://github.com/WordPress/wordpress-develop/blob/78cf2760354fe508d4abaf19403514034cbef1c9/src/wp-includes/rest-api/endpoints/class-wp-rest-widget-types-controller.php#L338

https://github.com/WordPress/wordpress-develop/blob/df8c3d0e089f77eb5c6fa971ba4ae9956255893e/src/wp-includes/rest-api/endpoints/class-wp-rest-widgets-controller.php#L729

File: src/wp-includes/rest-api/endpoints/class-wp-rest-taxonomies-controller.php [hunk 2]
```diff
@@ -272,16 +272,18 @@ public function prepare_item_for_response( $item, $request ) {
...
-				'collection'              => array(
-					'href' => rest_url( sprintf( '%s/%s', $this->namespace, $this->rest_base ) ),
-				),
-				'https://api.w.org/items' => array(
-					'href' => rest_url( rest_get_route_for_taxonomy_items( $taxonomy->name ) ),
-				),
-			)
-		);
+		if ( rest_is_field_included( '_links', $fields ) || rest_is_field_included( '_embedded', $fields ) ) {
+			$response->add_links(
```
Comment (spacedmonkey) [inline src/wp-includes/rest-api/endpoints/class-wp-rest-taxonomies-controller.php]: See https://github.com/WordPress/wordpress-develop/pull/1179/files#r899365716

**Comment by dlh01** [comment]
Will open a fresh PR against `trunk`.

**Pull request closed without merging**